feat(avm): add isolated test cases for DIV/FDIV divide by zero

MirandaWood opened the pull request:
Adds a missing test case where we have a divide by zero error and the result `c` is *not* set in tracegen. Discovered by David, thank you!

jeanmon commented on barretenberg/cpp/src/barretenberg/vm2/tracegen/alu_trace.test.cpp:
```diff
@@ -766,6 +809,40 @@ TEST_F(AluTraceGenerationTest, TraceGenerationFDivTagError)
+                                  ROW_FIELD_EQ(alu_ia, 6),
+                                  ROW_FIELD_EQ(alu_ib, 0),
+                                  ROW_FIELD_EQ(alu_ic, 0),
+                                  ROW_FIELD_EQ(alu_ia_tag, static_cast<uint8_t>(tag)),
```
tag is already casted into uint8_t above.

MirandaWood committed:
feat(avm): add isolated test cases for DIV/FDIV divide by zero

Adds a missing test case where we have a divide by zero error and the result `c` is *not* set in tracegen. Discovered by David, thank you!
